processor/transaction/payload.go

```diff
@@ -17,19 +17,22 @@ var (
 	processorSpanEntry  = common.MapStr{"name": processorName, "event": spanDocType}
 )
 
-type payload struct {
+type Payload struct {
 	Service m.Service
 	System  *m.System
 	Process *m.Process
 	User    *m.User
-	Events  []Event
+	Events  []*Event
```

Comment:
not clear why this, slice items are already references

Reply:
Because I am removing the references to the `Event` immediately after processing it. If garbage collection kicks in the Event can already be cleaned up.

https://github.com/elastic/apm-server/blob/83472af331ff13c45f62cd06769dca92a52fdec0/processor/transaction/payload.go#L92-L94